core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -883,20 +883,30 @@ public boolean hasChangeSetComputed() {
 
     @Override
     public EnvVars getEnvironment(TaskListener log) throws IOException, InterruptedException {
-        EnvVars env = super.getEnvironment(log);
```

Comment:
If super implementation changes, the changes won't be reflected here anymore. Probably it worth mentioning it in `Run#getEnvironment()` so that the users do not get confused

Reply:
Adressed